dags/process_elife_zip_dag.py

```diff
@@ -86,62 +146,31 @@ def extract_archived_files_to_bucket(**context) -> str:
 
         prefix = zip_file_name.replace('.zip', '/')
 
+        args = [(prefix, temp_zip_file.name, f) for f in ZipFile(temp_zip_file).namelist()]
+
         # extract zip to cloud bucket
-        for zipped_file_path in ZipFile(temp_zip_file).namelist():
-            # extract zipped files to disk to avoid using too much memory
-            with TemporaryFile(dir=TEMP_DIRECTORY) as temp_unzipped_file:
-                temp_unzipped_file.write(ZipFile(temp_zip_file).read(zipped_file_path))
-                temp_unzipped_file.seek(0)
-
-                s3_key = prefix + zipped_file_path
-                s3.put_object(
-                    Bucket=DESTINATION_BUCKET,
-                    Key=s3_key,
-                    Body=temp_unzipped_file
-                )
-                logger.info('%s uploaded to %s', s3_key, DESTINATION_BUCKET)
+        with ThreadPoolExecutor() as p:
```

Comment:
need to be able to configure the `max_workers` argument, to tweak the number of threads. The [default](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.ThreadPoolExecutor) seems to be `processors*5` which can be a lot.

Reply:
consider specifying `thread_name_prefix` which makes debugging easier; the Python threads are mapped on OS threads so they should be visible in `htop` with the right options

Reply:
updated.